analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantArgument.cs

```diff
@@ -34,34 +36,26 @@ public sealed class RedundantArgument : SonarDiagnosticAnalyzer
             context.RegisterNodeAction(
                 c =>
                 {
-                    // Can't use optional arguments in expression trees (CS0584), so skip those
-                    if (c.IsInExpressionTree())
+                    if (!c.IsInExpressionTree() // Can't use optional arguments in expression trees (CS0584), so skip those
+                        && ArgumentList(c) is { Arguments.Count: > 0 } argumentList
+                        && new CSharpMethodParameterLookup(argumentList, c.SemanticModel) is { MethodSymbol: { } } methodParameterLookup)
                     {
-                        return;
-                    }
-
-                    var argumentList = c.Node is InvocationExpressionSyntax invocationExpression
-                        ? invocationExpression.ArgumentList
-                        : ((ImplicitObjectCreationExpressionSyntaxWrapper)c.Node).ArgumentList;
-                    var methodParameterLookup = new CSharpMethodParameterLookup(argumentList, c.SemanticModel);
-
-                    if (methodParameterLookup.MethodSymbol != null)
-                    {
-                        foreach (var argumentMapping in methodParameterLookup.GetAllArgumentParameterMappings().Reverse().Where(x => x.Symbol.HasExplicitDefaultValue))
+                        foreach (var argumentMapping in methodParameterLookup.GetAllArgumentParameterMappings().Reverse().Where(x => ArgumentHasDefaultValue(x, c.SemanticModel)))
                         {
-                            var hasDefaultValue = ArgumentHasDefaultValue(argumentMapping, c.SemanticModel);
```

Comment:
I'm still not 100% sure about reverting a PR that we are not sure about why it was introduced in the first place. I suggest asking @pavel-mikula-sonarsource and if he's ok we can move on. With the current information, I think you are right, and even with methods with lengthy parameters, we should raise and suggest to use named arguments (see my [other comment](https://github.com/SonarSource/sonar-dotnet/pull/8222/files#r1368565462)).

Reply:
We got no feedback from @pavel-mikula-sonarsource until now. We either merge today before peach deployment or the PR will not be part of the release. There are only a few new issues in the ITs related to this change and the new issues are TPs (e.g. https://github.com/SonarSource/sonar-dotnet/pull/8222/files#r1365174847).

Reply:
This line was introduced as a fix to https://github.com/SonarSource/sonar-dotnet/issues/4620, because it was fixing annoying FP in the rule.